betamax-core/src/main/java/software/betamax/tape/MemoryTape.java

```diff
@@ -272,7 +274,14 @@ private void recordResponseBody(Response response, RecordedResponse recording) t
     private void recordBodyInline(Message message, RecordedMessage recording) throws IOException {
         boolean representAsText = isTextContentType(message.getContentType());
         if (representAsText) {
-            recording.setBody(message.getBodyAsText());
+            // TODO find a better way to accommodate for this behavior
+            if (message.getEncoding().equals("gzip")) {
+                recording.setBody(new GzipEncoder().decode(new ByteArrayInputStream(message.getBodyAsBinary())));
+            } else if (message.getEncoding().equals("deflate")) {
+                recording.setBody(new DeflateEncoder().decode(new ByteArrayInputStream(message.getBodyAsBinary())));
+            } else {
+                recording.setBody(message.getBodyAsText());
+            }
```

Comment:
`switch` is definitely better here and maybe you should have an enum for a encoding.